hub-saml/src/main/java/uk/gov/ida/saml/core/security/AssertionsDecrypters.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.ida.saml.core.security;
+
+import io.prometheus.client.Counter;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import uk.gov.ida.saml.security.AssertionDecrypter;
+import uk.gov.ida.saml.security.exception.SamlFailedToDecryptException;
+import uk.gov.ida.saml.security.validators.ValidatedResponse;
+import java.util.List;
+
+public class AssertionsDecrypters { 
+    private final List<AssertionDecrypter> assertionDecrypters;
+    
+    public AssertionsDecrypters(List<AssertionDecrypter> decrypters) {
+        this.assertionDecrypters = decrypters;
+    }
+    
+    public List<Assertion> decryptedAssertions(String className, ValidatedResponse response, Counter counter) {
+        
+        for (AssertionDecrypter assertionDecrypter : assertionDecrypters) {
+            try {
+                return assertionDecrypter.decryptAssertions(response);
+            } catch (SamlFailedToDecryptException e) {
+                counter.labels(response.getIssuer().getValue()).inc();
+                logFailedToDecryptWarning(className, response, e);
+            }
+        }
+        
+        throw samlFailedToDecryptException(className, response);
+    }
+
+    private void logFailedToDecryptWarning(String className, ValidatedResponse response, SamlFailedToDecryptException e) {
```

Comment:
Could we move `className` to the third arg as its "arguably" the least important param?
Also, could we consider:
1) changing the arg type to Class instead of String
2) changing arg name to `responseValidator`

These are pedantic, and willing to discuss.